### DISCUSSION TITLE
How do I log without suspend?

### DISCUSSION THREAD

**Discussion by Cyberavater** [klogging/klogging#242] [Q&A] [answered]
It seems every logging method in `Klogger` is `suspend`. Is it possible to log `non suspend` functions?

**Comment by mjstrasser** [comment, marked as answer]
`Klogger` is designed to work in coroutine scopes. Instead, you can use a `NoCoLogger` instance. It has functions with the same names that are not `suspend` functions.

You can create one by:

- implementing the interface `io.klogging.NoCoLogging`
- calling the function `noCoLogger()` in one of its forms

Let me know how you go. I will improve the [documentation](https://klogging.io/docs/loggers/defining-loggers) soon.

**Reply by Cyberavater** [reply]
Thanks, that was most helpful.
